src/rules/noInvalidVoidRule.ts

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
+        ruleName: "no-invalid-void",
+        description:
+            "Disallows usage of `void` type outside of return type. If `void` is used as return type, it shouldn't be a part of intersection/union type.",
```

Comment:
```suggestion
            "Disallows usage of `void` type outside of return type. If `void` is used as return type, it shouldn't be a part of 
 an intersection or union type.",
```

Reply:
It looks like your suggestion is a little bit invalid (syntax kind), I've done it with `dedent` in c80e995fd842d687e529ba2aded3e3e1abbbec9f